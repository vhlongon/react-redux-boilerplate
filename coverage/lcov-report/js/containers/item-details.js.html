<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/containers/item-details.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">js/containers</a> item-details.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {Component} from 'react';
import {connect} from 'react-redux';
&nbsp;
export const ItemDetails = (props) =&gt; {
  if(!props.activeItem) {
    return &lt;div&gt;Select an Item to get started.&lt;/div&gt;;
  } else {
    const {title, type} = props.activeItem;
    return (
      &lt;div&gt;
        &lt;h3&gt;Details for: &lt;/h3&gt;
        &lt;div&gt;Title: {title}&lt;/div&gt;
        &lt;div&gt;Pages: {type}&lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapStateToProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    activeItem: state.activeItem
  };
}
&nbsp;
export default connect(mapStateToProps)(ItemDetails);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sun Nov 06 2016 23:00:30 GMT+0100 (W. Europe Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
