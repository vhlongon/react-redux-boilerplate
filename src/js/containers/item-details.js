import React, {Component} from 'react';
import {connect} from 'react-redux';

export const ItemDetails = (props) => {
  if(!props.activeItem) {
    return <div>Select an Item to get started.</div>;
  } else {
    const {title, type} = props.activeItem;
    return (
      <div>
        <h3>Details for: </h3>
        <div>Title: {title}</div>
        <div>Pages: {type}</div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    activeItem: state.activeItem
  };
}

export default connect(mapStateToProps)(ItemDetails);
