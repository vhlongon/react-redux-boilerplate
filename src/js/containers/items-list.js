import React, {Component} from 'react';
import {connect} from 'react-redux';
import {selectItem} from '../actions/index';

export class ItemsList extends Component {

  renderList = () => {
    return this.props.items.map((item) => {
        return (
          <li
            key={item.title}
            onClick={() => this.props.selectItem(item)}
            >
            <span>title: {item.title}; </span>
            <span>type: {item.type}</span>
          </li>
        );
    });

  }

  render = () => {
    return (
      <div>
        {this.renderList()}
      </div>
    );
  }
}

// USED FOR BINDING REDUCERS
function mapStateToProps(state) {
  return {
    items: state.items
  };
}


// Promote Component from a normal React component to a container -
// this glue together the react view to the redux state object
// it needs to know about this new dispatch method, selectItem. Make it available as a prop
export default connect(mapStateToProps, {selectItem})(ItemsList);
