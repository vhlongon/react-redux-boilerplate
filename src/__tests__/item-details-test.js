import React from 'react';
import {ItemDetails} from '../js/containers/item-details';
import renderer from 'react-test-renderer';
import {mockActiveItem} from './mock';

// This example uses react-test-renderer
describe('components', () => {
  describe('ItemDetails', () => {
    test('Should show the initial state / no item selected', () => {
      const component = renderer.create(
          <ItemDetails activeItem={null} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('Should show a selected/clicked item', () => {
      const component = renderer.create(
          <ItemDetails activeItem= {mockActiveItem} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
