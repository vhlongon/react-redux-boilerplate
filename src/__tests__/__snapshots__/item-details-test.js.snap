exports[`components ItemDetails Should show a selected/clicked item 1`] = `
<div>
  <h3>
    Details for: 
  </h3>
  <div>
    Title: 
    Deus Ex-Machine
  </div>
  <div>
    Pages: 
    game
  </div>
</div>
`;

exports[`components ItemDetails Should show the initial state / no item selected 1`] = `
<div>
  Select an Item to get started.
</div>
`;
